custom-vu/src/main/kotlin/jces1209/vu/action/boards/ViewBoard.kt

```diff
@@ -0,0 +1,121 @@
+import jces1209.vu.page.JiraTips
+import jces1209.vu.page.boards.view.BoardContent
+import jces1209.vu.page.boards.view.BoardPage
+import jces1209.vu.page.boards.view.cloud.CloudKanbanBoardPage
+import jces1209.vu.page.boards.view.cloud.CloudNextGenBoardPage
+import jces1209.vu.page.boards.view.cloud.CloudScrumBacklogPage
+import jces1209.vu.page.boards.view.cloud.CloudScrumSprintPage
+import jces1209.vu.page.contextoperation.ContextOperationBoard
+import jces1209.vu.utils.boards.BoardsFrequencyManager
+import org.apache.logging.log4j.LogManager
+import org.apache.logging.log4j.Logger
+import org.openqa.selenium.WebDriver
+
+abstract class ViewBoard(
+    private val driver: WebDriver,
+    private val measure: Measure,
+    private val issueKeyMemory: IssueKeyMemory,
+    private val viewBoardMeasureType: ActionType<Unit>,
+    private val issuePreviewMeasureType: ActionType<Unit>,
+    private val contextOperationMeasureType: ActionType<Unit>,
+    private val configureBoardMeasureType: ActionType<Unit>,
+    private val viewIssueProbability: Float,
+    private val configureBoardProbability: Float,
+    private val contextOperationProbability: Float
+) {
+    protected val logger: Logger = LogManager.getLogger(this::class.java)
+    protected val jiraTips = JiraTips(driver)
+
+    protected fun viewBoard(measureType: ActionType<Unit>, board: BoardPage): BoardContent? {
+        return measure.measure(VIEW_BOARD) {
+            measure.measure(measureType,
+                isSilent = false,
+                observation = {
+                    issueKeyMemory.remember(it.getIssueKeys())
+                    IssuesOnBoard(it.getIssueCount()).serialize()
+                }
+            ) {
+                board
+                    .goToBoard()
+                    .waitForBoardPageToLoad()
+            }
+        }
+    }
+
+    protected fun contextOperation(measureType: ActionType<Unit>) {
+        measure.measure(MeasureType.CONTEXT_OPERATION_BOARD, contextOperationProbability) {
+            measure.measure(measureType, isSilent = false) {
+                ContextOperationBoard(driver)
+                    .open()
+            }
+        }
+            ?.close()
+    }
+
+    protected fun previewIssue(measureType: ActionType<Unit>, board: BoardPage) {
+        jiraTips.closeTips()
+        measure.measure(MeasureType.ISSUE_PREVIEW_BOARD) {
+            measure.measure(measureType, isSilent = false) {
+                board.previewIssue()
+            }
+        }
+            ?.closePreviewIssue()
+    }
+
+    protected fun configureBoard(measureType: ActionType<Unit>, board: BoardPage) {
+        measure.measure(MeasureType.CONFIGURE_BOARD, configureBoardProbability) {
+            measure.measure(measureType, isSilent = false) {
+                board
+                    .configureBoard()
+                    .waitForLoadPage()
+            }
+        }
+    }
+
+    protected fun viewBoardContent(board: BoardPage) {
```

Comment:
Let's rename this method - it doesn't make only view board
e.g. `generalRun`

Reply:
What about `runBoardActions`?